Bogus unrelated --explain text for "error[E0507]: cannot move out of captured outer variable in an `Fn` closure"

`rustc --explain E0507` prints a message which apparently discusses "error[E0507]: cannot move out of borrowed content":
> You tried to move out of a value which was borrowed. ...

(That's funny, because the error said I tried to "move out of captured outer variable in an `Fn` closure".)

  * If "cannot move out of captured outer variable in an `Fn` closure" really _is_ the same error, **this should be explained in the `--explain` text.**
  * If it is _not_ the same error, it should have its own error code and explanation.

--- a/src/librustc_mir/error_codes.rs
+++ b/src/librustc_mir/error_codes.rs
@@ -1646,7 +1646,14 @@ fn print_fancy_ref(fancy_ref: &FancyNum){
 "##,
 
 E0507: r##"
-You tried to move out of a value which was borrowed. Erroneous code example:
+You tried to move out of a value which was borrowed.
+
+This can also happen when using a type implementing `Fn` or `FnMut`, as neither
+allows moving out of them (they usually represent closures which can be called
+more than once). Much of the text following applies equally well to non-`FnOnce`
+closure bodies.
+
+Erroneous code example:
 
 ```compile_fail,E0507
 use std::cell::RefCell;

